truffle/src/main/java/org/jruby/truffle/nodes/core/BigDecimalNodes.java

```diff
@@ -109,22 +168,25 @@ public RubyBasicObject initialize(RubyBasicObject self, RubyBasicObject v) {
         }
 
         @Specialization(guards = "isRubyString(v)")
+        @CompilerDirectives.TruffleBoundary
         public RubyBasicObject initializeFromString(RubyBasicObject self, RubyBasicObject v) {
-            // TODO (pitr 20-May-2015): add NaN, Infinity handling
+            // TODO (pitr 21-May-2015): add profile?
+            // TODO (pitr 25-May-2015): construction of positive and negative zero
             switch (v.toString()) {
```

Comment:
In short: badly :disappointed:. Just saw https://github.com/pitr-ch/jruby/blob/bigdecimal/truffle/src/main/java/org/jruby/truffle/runtime/core/RubyBasicObject.java#L175-175 so I have to fix this. Thanks.